src/parser.cc

```diff
@@ -773,20 +773,28 @@ bool readFileInternal(const std::string &_filename, const bool _convert,
   {
     return true;
   }
-  else if (URDF2SDF::IsURDF(filename))
+  else
   {
-    URDF2SDF u2g;
-    auto doc = makeSdfDoc();
-    u2g.InitModelFile(filename, _config, &doc);
-    if (sdf::readDoc(&doc, _sdf, filename, _convert, _config, _errors))
+    tinyxml2::XMLElement *robotXml = xmlDoc.FirstChildElement("robot");
+    if (robotXml && URDF2SDF::IsURDF(filename))
     {
-      sdfdbg << "parse from urdf file [" << _filename << "].\n";
-      return true;
+      URDF2SDF u2g;
+      auto doc = makeSdfDoc();
+      u2g.InitModelFile(filename, _config, &doc);
+      if (sdf::readDoc(&doc, _sdf, filename, _convert, _config, _errors))
+      {
+        sdfdbg << "parse from urdf file [" << _filename << "].\n";
+        return true;
+      }
+      else
+      {
+        sdferr << "parse as old deprecated model file failed.\n";
```

Comment:
This error message is not really correct. Not 100% sure what it should say. Maybe "Failed to parse the URDF file after converting to SDFormat"?

Reply:
I agree this is confusing although I was trying to keep the scope of this PR focused on the "fix" of removing the "robot" related message, dunno if this is comes as part of it or it should go into `main` as a behavior change, what do you think?

Reply:
Done in https://github.com/gazebosim/sdformat/pull/1290/commits/41bb5bd0a1e29442d84cad4503677c9d4286e9ac.